2016/proposals/singh-shubham.md

```diff
@@ -0,0 +1,210 @@
+Software Carpentary is a step in this direction.
+Since 1998, Software Carpentry has been teaching researchers in science, engineering, medicine, and related disciplines the computing skills they need to get more done in less time and with less pain.
+Software Carpentary , uses AMY ( a django based web application) to to keep track of who's qualified to teach, what workshops have run, who taught and attended. As more and more people are applying to become instructors at Software Carpentary. It has become essential to manage the workflow for instructor training.
+
+This project aims at designing such a workflow and implementing it by adding appropriate views to AMY.
+
+## Proposed workflow
+
+![HTML MOCKUP flowchart](https://cloud.githubusercontent.com/assets/15982349/13770094/aa23d670-eaaa-11e5-81a8-312f860ecd7b.png)
+
+1. Interested Trainees apply as Group or Individuals by filling forms.
+![HTML MOCKUP FORMS](https://cloud.githubusercontent.com/assets/15982349/13816694/6c7a68e6-ebb5-11e5-809e-eb5bd00d7a22.png)
+2. All applications are listed in the *ALL_APPLICATIONS* view. In this view the lead trainer can accept or reject an application.
+The lead trainer has to compose one sample email by filling a web-form before accepting an application, this email will contain information such as
+date, location, mode(online or in person), instructor name and instructor email id. This information is also stored in the database for the accepted applicant to be queried on later.
+Accepting an application will send an email, to the applicant that he/she has been accepted into the program with other details about the actual training along with one randomly generated password to login in to their account (this password can be changed to anything by the trainee after the first login).
+The sample email has to be written once and can be sent to all the people who have been accepted for a particular training session.
+For another training session the lead trainer would just have to fill the web-form again, overwriting previous values.
+![HTML MOCKUP ALL_APPLICATIONS]( https://cloud.githubusercontent.com/assets/15982349/13816688/66a3e7e4-ebb5-11e5-83f1-5086ddddac13.png)
+
+3. All accepted applications, are then transferred to the Trainee view. In this view the lead trainer, trainers,discussion leaders, lesson maintainers can track progress of the trainees, provide feedback and update the status of every stage. They are also notified here if the Trainee has requested a discussion session.
+This list can be filtered on the following criteria:
+  * instructor assigned.
+  * start date of training.
+  * location of the training.
+
+  When the training is completed and the trainee has been certified, the lead instructor can export the trainee as instructor, to move their accounts to the Person table, and remove them from the Trainee table.
```

Comment:
I think everyone should go in the Person table right from the start - in particular, many trainees will be alumni of our regular workshops, so we will already have information about them.  The Trainee record should then have a foreign key to that Person information to avoid duplication.